### PR TITLE
Disable Python linting by default (fixes #1789)

### LINKED ISSUE #1789: [Bug]: Remove linting from CodeAct by default
### Is there an existing issue for the same bug?

- [X] I have checked the troubleshooting document at https://opendevin.github.io/OpenDevin/modules/usage/troubleshooting
- [X] I have checked the existing issues.

### Describe the bug

CodeAct comes with some of SWEAgent's opinions on how code should be formatted. It makes assumptions about (a) how our users might want to format code and (b) what sort of code they might be working with (e.g. it mangled some JS indentation here: https://github.com/OpenDevin/OpenDevin/issues/1710)

While the linting might be helpful for the SWE-bench benchmark specifically, we should disable it for end-users

### Current Version

```bash
0.5
```

### Installation and Configuration

```bash
docker run
```

### Model and Agent

CodeAct

### Reproduction Steps

Have CodeAct work in JavaScript

### Logs, Errors, Screenshots, and Additional Context

_No response_

### PR DESCRIPTION
**Pull request by isavita**
## Description
This PR addresses [Issue #1789](https://github.com/OpenDevin/OpenDevin/issues/1789) by modifying the edit function to disable Python linting by default. Linting can be enabled by setting the `LINT_ENABLED` environment variable to any non-empty value before running the edit function.

### How to Enable Linting
```bash
export RUN_LINT_AFTER_EDIT=1
edit 10:20
```
Or
```bash
export RUN_LINT_AFTER_EDIT=true
edit 10:20
```
By default, linting is disabled to prevent unnecessary syntax checks.

### CODE REVIEW

**Review by li-boxuan — COMMENTED** [review]
Thanks for the fix! There's another one in `cursors_edit_linting.sh`. I think we should also expose this to the frontend - don't have to include it in this PR if you are not familiar with frontend.

**Comment by isavita** [comment]
@li-boxuan done here 04333c0d094abb65784e5a68459ac1d1b9d082e5

File: opendevin/runtime/plugins/swe_agent_commands/cursors_edit_linting.sh [hunk 1]
```diff
@@ -35,8 +35,8 @@ edit() {
     local new_lines=("${lines[@]:0:$start_line}" "${replacement[@]}" "${lines[@]:$((end_line))}")
     # Write the new stuff directly back into the original file
     printf "%s\n" "${new_lines[@]}" >| "$CURRENT_FILE"
-    # Run linter
-    if [[ $CURRENT_FILE == *.py ]]; then
+    # Run linter if enabled
+    if [[ $CURRENT_FILE == *.py && -n "$LINT_ENABLED" ]]; then
```
Comment (li-boxuan) [inline opendevin/runtime/plugins/swe_agent_commands/cursors_edit_linting.sh:39]: let's make the naming a bit clearer

```suggestion
    if [[ $CURRENT_FILE == *.py && -n "$RUN_LINT_AFTER_EDIT" ]]; then
```

File: opendevin/runtime/plugins/swe_agent_commands/edit_linting.sh [hunk 2]
```diff
@@ -67,8 +67,8 @@ edit() {
     # Write the new stuff directly back into the original file
     printf "%s\n" "${new_lines[@]}" >| "$CURRENT_FILE"
 
-    # Run linter
-    if [[ $CURRENT_FILE == *.py ]]; then
+    # Run linter if enabled
+    if [[ $CURRENT_FILE == *.py && -n "$LINT_ENABLED" ]]; then
```
Comment (li-boxuan) [inline opendevin/runtime/plugins/swe_agent_commands/edit_linting.sh:71]:
```suggestion
    if [[ $CURRENT_FILE == *.py && -n "$RUN_LINT_AFTER_EDIT" ]]; then
```

**Review by li-boxuan — APPROVED** [review]
LGTM

**Comment by li-boxuan** [comment]
@xingyaoww This might affect the benchmark for CodeAct. The linting is turned off by default and can be enabled via `RUN_LINT_AFTER_EDIT=true`

**Comment by li-boxuan** [comment]
Not sure making it a config would work - we need to somehow feed this config into shell scripts.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by li-boxuan** [comment]
I am a little bit worried that this might be overlooked during benchmarking and negatively affect the results. How about we hold this PR until the current round of benchmarking is finished?

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by neubig** [comment]
@li-boxuan do you want to remove your approval then?

**Comment by neubig** [comment]
Sounds good!

**Review by li-boxuan — APPROVED** [review]
Thank you! I'll check-in https://github.com/OpenDevin/OpenDevin/pull/1861 right after this, to preserve your authorship.

**Pull request merged**